src/steamship/agents/schema/memory_strategies.py

```diff
@@ -0,0 +1,73 @@
+from typing import List
+
+import tiktoken
+
+from steamship import Block
+from steamship.agents.memory.chathistory import ChatHistory, MessageSelector
+from steamship.data.tags.tag_constants import RoleTag
+
+
+class NoMemory(MessageSelector):
+    def get_messages(self, chat_history: ChatHistory) -> List[Block]:
+        return []
+
+
+def is_user_message(block: Block) -> bool:
+    role = block.chat_role
+    return role == RoleTag.USER
+
+
+def is_assistant_message(block: Block) -> bool:
+    role = block.chat_role
+    return role == RoleTag.ASSISTANT
+
+
+class MessageWindowMessageSelector(MessageSelector):
+    k: int
+    # k: Field(default=4, ge=1, description="Number of message pairs to return from history. A message pair is a
+    # single set of messages exchanged between a user and an assistant."
+
+    def get_messages(self, chat_history: ChatHistory) -> List[Block]:
+        all_messages = chat_history.messages
+        all_messages.pop()  # don't add the current prompt to the memory
```

Comment:
@dkolas do we think this hints at some "flaw" in the system?  I couldn't decide when first working through this.